syft/pkg/cataloger/deb/parse_dpkg_status.go

```diff
@@ -48,20 +51,54 @@ func parseDpkgStatus(reader io.Reader) ([]pkg.Package, error) {
 }
 
 // parseDpkgStatusEntry returns an individual Dpkg entry, or returns errEndOfPackages if there are no more packages to parse from the reader.
-// nolint:funlen
-func parseDpkgStatusEntry(reader *bufio.Reader) (entry pkg.DpkgMetadata, err error) {
-	dpkgFields := make(map[string]interface{})
+func parseDpkgStatusEntry(reader *bufio.Reader) (pkg.DpkgMetadata, error) {
 	var retErr error
+	dpkgFields, err := extractAllFields(reader)
+	if err != nil {
+		if !errors.Is(err, errEndOfPackages) {
+			return pkg.DpkgMetadata{}, err
+		}
+		retErr = err
+	}
+
+	var entry pkg.DpkgMetadata
+	err = mapstructure.Decode(dpkgFields, &entry)
+	if err != nil {
+		return pkg.DpkgMetadata{}, err
+	}
+
+	name, version := extractSourceVersion(entry.Source)
+	if version != "" {
+		entry.SourceVersion = version
+		entry.Source = name
+	}
+
+	// there may be an optional conffiles section that we should persist as files
+	if conffilesSection, exists := dpkgFields["Conffiles"]; exists && conffilesSection != nil {
+		if sectionStr, ok := conffilesSection.(string); ok {
+			entry.Files = parseDpkgConffileInfo(strings.NewReader(sectionStr))
+		}
+	}
+
+	if entry.Files == nil {
+		// ensure that the collection is always allocated
+		entry.Files = make([]pkg.DpkgFileRecord, 0)
```

Comment:
This is surprising... isn't there a way for this to be defined beforehand instead of having to check like this?

Reply:
Ohhh I guess not because the struct wasn't initialized. Shouldn't there be a constructor for this?

Reply:
Right, we don't have a constructor here (or in many [any?] of the underlying pkg.*Metadata structs). We typically have a `parse*()` function that is acting like the constructor for an object. I'll move this line to where `entry` is created to make this more straight forward 👍